### ISSUE TITLE
List index out of range when adding vpool on KLONE cluster 3

### ISSUE THREAD

**Issue by pploegaert** [openvstorage/framework#790]
Note that there is no DB role assigned on a compute node, only on the controller node

diskpartition_info content =

```
2016-08-09 12:51:05 71700 +0200 - cmp03 - 5505/140148209542976 - lib/storagerouter - 10 - WARNING - {'READ': [{'available': 93242880000, 'in_use': False, 'storagerouter_guid': u'e747507e-bae3-4772-b86e-d1910fa554cd', 'mountpoint': u'/mnt/ssd1', 'ssd': True, 'guid': 'd58c0c1d-e608-49c7-93ed-0451489556af', 'size': 100028907520L}], 'WRITE': [{'available': 224072503296, 'in_use': False, 'storagerouter_guid': u'e747507e-bae3-4772-b86e-d1910fa554cd', 'mountpoint': u'/mnt/ssd2', 'ssd': True, 'guid': '858b8926-8580-4986-94cd-d11099ba1796', 'size': 240055746560L}], 'DB': [], 'SCRUB': [{'available': 224072503296, 'in_use': False, 'storagerouter_guid': u'e747507e-bae3-4772-b86e-d1910fa554cd', 'mountpoint': u'/mnt/ssd3', 'ssd': True, 'guid': 'cd824ab3-ad17-4de7-9dea-5e112b37eeed', 'size': 240055746560L}], 'BACKEND': []}
```

```
2016-08-09 11:09:48 11400 +0200 - cmp03 - 4461/139852145686336 - celery/celery.worker.job - 657 - ERROR - Task ovs.storagerouter.add_vpool[cdf1bd67-37ba-41d1-a6e8-9e3b0e831f2e] raised unexpected: IndexError('list index out of range',)
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/OpenvStorage/ovs/lib/storagerouter.py", line 485, in add_vpool
    db_available_size = partition_info[DiskPartition.ROLES.DB][0]['available']
IndexError: list index out of range
```

**Comment by khenderick** [comment]
A DB role is mandatory on every node.

The wizard should validate this (so that's a todo - why didn't the wizard complain here)
The add_vpool logic also checks this, but it tries to run all checks before raising a list of all issues found, and the above stacktrace is on one of these subsequent checks.

**Comment by khenderick** [comment]
In the meantime, the code will check for the DB role before starting extending/adding the vPool.

**Comment by khenderick** [comment]
Fixed by #1038, packaged in openvstorage-2.7.4-rev.4147.6f02d53

**Comment by JeffreyDevloo** [comment]
### Steps
- Setup cluster
- Added 1 node with DB role
- Created backend
- Setup vpool
- Tried to extend vpool

### Output
GUI
![selection_102](https://cloud.githubusercontent.com/assets/17570109/21386220/1cd74248-c772-11e6-9d64-68261b45cf97.png)
API
```
expected: ValueError('Errors validating the specified parameters:\n - Too much space requested for WRITE cache. Available: 0.00 GiB, Requested: 49.00 GiB\n - At least 1 DB partition role is required\n - At least 1 WRITE partition role is required',)
Dec 21 13:21:38 ovs-node-1 celery[1899]: Traceback (most recent call last):
Dec 21 13:21:38 ovs-node-1 celery[1899]:   File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
Dec 21 13:21:38 ovs-node-1 celery[1899]:     R = retval = fun(*args, **kwargs)
Dec 21 13:21:38 ovs-node-1 celery[1899]:   File "/usr/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
Dec 21 13:21:38 ovs-node-1 celery[1899]:     return self.run(*args, **kwargs)
Dec 21 13:21:38 ovs-node-1 celery[1899]:   File "/opt/OpenvStorage/ovs/lib/storagerouter.py", line 369, in add_vpool
Dec 21 13:21:38 ovs-node-1 celery[1899]:     raise ValueError('Errors validating the specified parameters:\n - {0}'.format('\n - '.join(set(error_messages))))
Dec 21 13:21:38 ovs-node-1 celery[1899]: ValueError: Errors validating the specified parameters:
Dec 21 13:21:38 ovs-node-1 celery[1899]:  - Too much space requested for WRITE cache. Available: 0.00 GiB, Requested: 49.00 GiB
Dec 21 13:21:38 ovs-node-1 celery[1899]:  - At least 1 DB partition role is required
Dec 21 13:21:38 ovs-node-1 celery[1899]:  - At least 1 WRITE partition role is required

```
And my vpool is no longer usable.

### Test result
Test failed.

**Comment by JeffreyDevloo** [comment]
This ticket handled mostly over the GUI. Therefore closing.
